src/app/services/member-contract.service.ts

```diff
@@ -35,161 +35,130 @@ export class MemberContractService {
-        }
-        resolve(res);
-      });
-    }) as Promise<string>;
+    return this.membersContract.methods.isRegularOrBoardMember(_memberAddress).call();
   }
 
   // check if board member has already confirmed applicant
   async hasConfirmedApplicant(_boardMemberAddress: string, _applicantAddress: string): Promise<boolean> {
-    return new Promise((resolve, reject) => {
-      this.membersContract.hasConfirmedApplicant.call(_boardMemberAddress, _applicantAddress, function (err, res) {
-        if (err != null) {
-          reject(err);
-        }
-        resolve(res);
-      });
-    }) as Promise<boolean>;
+  return this.membersContract.methods.hasConfirmedApplicant(_boardMemberAddress, _applicantAddress).call();
   }
 
   /* Contract Transactions */
 
   // apply for Membership
   async applyForMembership(_name: string) {
-    this.membersContract.applyForMembership.sendTransaction(_name, function (err) {
-      if (err) {
-        console.log(err);
-      }
-    });
+    const acc = await this._web3Service.getAccount();
+    this.membersContract.methods.applyForMembership(_name).send({from: acc});
   }
 
   // change Name
   async changeName(_newName: string) {
-    this.membersContract.changeName.sendTransaction(_newName, function (err) {
-      if (err) {
-        console.log(err);
-      }
-    });
+    const acc = await this._web3Service.getAccount();
+    this.membersContract.methods.changeName(_newName).send({from: acc});
   }
 
   // confirm Application of applicant
   async confirmApplication(_applicantAddress: string, callback: Function) {
-    this.membersContract.confirmApplication.sendTransaction(_applicantAddress, callback);
+    const acc = await this._web3Service.getAccount();
+    this.membersContract.methods.confirmApplication(_applicantAddress).send({from: acc}, callback);
   }
 
   // resign membership of logged in member
   async resignOwnMembership() {
-    this.membersContract.resignOwnMembership.sendTransaction(function (err) {
-      if (err) {
-        console.log(err);
-      }
-    });
+    const acc = await this._web3Service.getAccount();
+    this.membersContract.methods.resignOwnMembership().send({from:acc});
   }
 
   // Initially set voting contract address
   async setVotingContractAddress(_votingContractAdrres: string) {
-    this.membersContract.setVotingContractAddress.sendTransaction(_votingContractAdrres, function (err) {
-      if (err) {
-        console.log(err);
-      }
-    });
+    const acc = await this._web3Service.getAccount();
+    this.membersContract.methods.setVotingContractAddress(_votingContractAdrres).send({from: acc})
   }
 
   /* /Contract Functions */
 
   /* Contract Events */
   getMemberAppliedEvent() {
-    return this.membersContract.MemberApplied();
+    return this.membersContract.events.MemberApplied({
+      filter: { myIndexedParam: [20, 23], myOtherIndexedParam: '0x123456789...' },
+      fromBlock: 0
+    }, function (error, event) { console.log(event); })
```

Comment:
Looks like the event is only logged, I think with the new web3.js version the function should take a callback as a parameter that does something useful with the event

Reply:
>  something useful with the event

Are you referring to something like a success message for the user?
e. g. "You(Member) successfully applied"

Reply:
There is already some frontend logic in place that refreshes certain content based on the events (e.g. update the member list of a member status changed). Best check what the previous version of the code did with the result of the events, and try to apply the same with the new/changed web3j library.

Reply:
Resolved with: commit 82b09d21fdba4ef6683d8f5dd4b7af0f0f77bed7